ts/src/bitget.ts

```diff
@@ -3973,52 +4050,82 @@ export default class bitget extends Exchange {
         await this.loadMarkets ();
         const market = this.market (symbol);
         let marketType = undefined;
+        let marginMode = undefined;
+        let response = undefined;
         [ marketType, params ] = this.handleMarketTypeAndParams ('fetchCanceledAndClosedOrders', market, params);
-        const endTime = this.safeIntegerN (params, [ 'endTime', 'until', 'till' ]);
-        params = this.omit (params, [ 'until', 'till' ]);
+        [ marginMode, params ] = this.handleMarginModeAndParams ('fetchCanceledAndClosedOrders', params);
+        const symbolRequest = (marginMode !== undefined) ? (market['info']['symbolName']) : (market['id']);
         const request = {
-            'symbol': market['id'],
+            'symbol': symbolRequest,
         };
-        if (since !== undefined) {
-            request['startTime'] = since;
-        }
-        let method = this.getSupportedMapping (marketType, {
-            'spot': 'privateSpotPostTradeHistory',
-            'swap': 'privateMixGetOrderHistory',
-            'future': 'privateMixGetOrderHistory',
-        });
+        const now = this.milliseconds ().toString ();
+        const endTime = this.safeIntegerN (params, [ 'endTime', 'until', 'till' ]);
         const stop = this.safeValue (params, 'stop');
-        if (stop) {
-            if (marketType === 'spot') {
-                method = 'privateSpotPostPlanHistoryPlan';
-            } else {
-                method = 'privateMixGetPlanHistoryPlan';
-            }
-            params = this.omit (params, 'stop');
-        }
-        if (marketType === 'swap' || stop) {
+        params = this.omit (params, [ 'until', 'till', 'stop' ]);
+        if (stop || (marketType === 'swap') || (marketType === 'future')) {
             if (limit === undefined) {
                 limit = 100;
             }
             request['pageSize'] = limit;
             if (since === undefined) {
-                since = 0;
+                if (marketType === 'spot') {
+                    const ninetyDaysMilliseconds = '7776000000';
+                    const sinceString = Precise.stringSub (now, ninetyDaysMilliseconds);
+                    since = this.parseNumber (sinceString);
```

Comment:
@Dan-krm  We can never parse timestamps like this, it will result in a float in python

Reply:
```
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 1960, in fetch2
    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 221, in fetch
    self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/bitget.py", line 5472, in handle_errors
    raise ExchangeError(feedback)  # unknown message
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ccxt.base.errors.ExchangeError: bitget {"code":"40020","msg":"Parameter startTime error","requestTime":1697792150030,"data":null}
```